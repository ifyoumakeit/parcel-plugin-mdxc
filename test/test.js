const path = require("path");
const MarkdownAsset = require("../MarkdownAsset");
const Parser = require("parcel-bundler/src/Parser");

describe("MarkdownAsset", () => {
  it("should parse markdown correctly.", async () => {
    const asset = new MarkdownAsset(path.resolve(__dirname, "./index.md"), {
      rootDir: __dirname,
    });
    const processed = await asset.process();
    expect(processed).toMatchSnapshot();
  });
  it("should parse markdown with images correctly.", async () => {
    const asset = new MarkdownAsset(path.resolve(__dirname, "./index.img.md"), {
      rootDir: __dirname,
      parser: new Parser(),
      publicURL: "/",
    });
    const processed = await asset.process();
    expect(processed).toMatchSnapshot();
  });
});
