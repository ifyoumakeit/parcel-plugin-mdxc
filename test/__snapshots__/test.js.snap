// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MdxcAsset should parse markdown correctly. 1`] = `
Array [
  Object {
    "type": "js",
    "value": "import React, { createElement, createFactory } from 'react'
import Mdx from \\"./index.md\\";
export default function({ factories={} }) {
  const {
    p = createFactory('p'),
    wrapper = createFactory('div'),
  } = factories

  return wrapper({},

p({},
  \\"import React from \\\\\\"react\\\\\\";\\\\nimport ReactDOM from \\\\\\"react-dom\\\\\\";\\",
),
p({},
  \\"ReactDOM.render(\\\\n  \\",
  createElement(\\"h1\\", null,
    \\"\\\\n    \\",
    createElement(Mdx, null),
    \\"Garwacke\\\\n  \\",
  ),
  \\",\\\\n  document.querySelector(\\\\\\"#root\\\\\\")\\\\n);\\",
)

  )
}
",
  },
]
`;

exports[`MdxcAsset should parse markdown with images correctly. 1`] = `
Array [
  Object {
    "type": "js",
    "value": "import React, { createElement, createFactory } from 'react'

export default function({ factories={} }) {
  const {
    h1 = createFactory('h1'),
    h2 = createFactory('h2'),
    img = createFactory('img'),
    p = createFactory('p'),
    wrapper = createFactory('div'),
  } = factories

  return wrapper({},

h1({\\"id\\": \\"Test\\"},
  \\"Test\\",
),
h2({\\"id\\": \\"H2\\"},
  \\"H2\\",
),
p({},
  img({\\"src\\": \\"./img.jpg\\", \\"alt\\": \\"\\"}),
)

  )
}
",
  },
]
`;
